fastplotlib/widgets/image.py

```diff
@@ -372,149 +360,29 @@ def __init__(
-                    )
-                self._slider_dims: List[str] = [self.dims_order[0][slider_dims]]
-
-            # if dimension specified by str
-            elif isinstance(slider_dims, str):
-                if slider_dims not in self.dims_order[0]:
-                    raise ValueError(
-                        f"if `slider_dims` is a <str>, it must be a character found in `dims_order`. "
-                        f"Your `dims_order` characters are: {set(self.dims_order[0])}."
-                    )
-                self._slider_dims: List[str] = [slider_dims]
-
-        # multiple sliders, one for each dimension
-        elif isinstance(slider_dims, list):
-            self._slider_dims: List[str] = list()
-
-            # make sure window_funcs and frame_apply are dicts if multiple sliders are desired
-            if (not isinstance(window_funcs, dict)) and (window_funcs is not None):
-                raise TypeError(
-                    f"`window_funcs` must be a <dict> if multiple `slider_dims` are provided. You must specify the "
-                    f"window for each dimension."
-                )
-            if (not isinstance(frame_apply, dict)) and (frame_apply is not None):
-                raise TypeError(
-                    f"`frame_apply` must be a <dict> if multiple `slider_dims` are provided. You must specify a "
-                    f"function for each dimension."
-                )
-
-            for sdm in slider_dims:
-                if isinstance(sdm, int):
-                    ao = np.array([v for v in self.dims_order])
-                    if not np.all(ao == ao[0]):
-                        raise ValueError(
-                            f"`dims_order` for all arrays must be identical if passing in a <int> `slider_dims` argument. "
-                            f"Pass in a <str> argument if the `dims_order` are different for each array."
-                        )
-                    # parse int to a str
-                    self.slider_dims.append(self.dims_order[0][sdm])
-
-                elif isinstance(sdm, str):
-                    if sdm not in self.dims_order[0]:
-                        raise ValueError(
-                            f"if `slider_dims` is a <str>, it must be a character found in `dims_order`. "
-                            f"Your `dims_order` characters are: {set(self.dims_order[0])}."
-                        )
-                    self.slider_dims.append(sdm)
-
-                else:
-                    raise TypeError(
-                        "If passing a list for `slider_dims` each element must be either an <int> or <str>"
-                    )
-
-        else:
-            raise TypeError(
-                f"`slider_dims` must a <int>, <str> or <list>, you have passed a: {type(slider_dims)}"
-            )
+        # Sliders are made for all dimensions except the last 2
+        self._slider_dims = list()
+        for dim in range(self.ndim - 2):
```

Comment:
I think you can do something like:

```python
for dim_ix, dim_name in zip(range(self.ndim - 2), ("t", "z"))
```

Reply:
@kushalkolar I think we should move away from self.ndim in general - as we add support for things like RGBA, keeping track of the "2" will get annoying. This is nicer imo: 

`
self._slider_dims = list()
for dim in range(self.ndim):
    if dim in ALLOWED_SLIDER_DIMS.keys():
        self.slider_dims.append(ALLOWED_SLIDER_DIMS[dim])
`

Reply:
ah right, so then it'll stay restricted to just t and z sliders